src/fsw/FCCode/ADCSBoxMonitor.cpp

```diff
@@ -8,24 +8,52 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
     unsigned int offset, Devices::ADCS &_adcs)
     : TimedControlTask<void>(registry, "adcs_monitor", offset),
     adcs_system(_adcs),
-    rwa_speed_rd_sr(adcs::rwa::min_speed_read, adcs::rwa::max_speed_read, 8*3), //full res is 16*3
+    rwa_speed_rd_sr(adcs::rwa::min_speed_read, adcs::rwa::max_speed_read, 16*3), //referenced from I2C_Interface.doc
+    rwa_speed_rd_component_sr(adcs::rwa::min_speed_read,adcs::rwa::max_speed_read, 8),
     rwa_speed_rd_f("adcs_monitor.rwa_speed_rd", rwa_speed_rd_sr),
-    rwa_torque_rd_sr(adcs::rwa::min_torque, adcs::rwa::max_torque, 8*3), //full res is 16*3
+    rwa_speed_rd_x_f("adcs_monitor.rwa_speed_rd.x",rwa_speed_rd_component_sr),
+    rwa_speed_rd_y_f("adcs_monitor.rwa_speed_rd.y",rwa_speed_rd_component_sr),
+    rwa_speed_rd_z_f("adcs_monitor.rwa_speed_rd.z",rwa_speed_rd_component_sr),
+
+    rwa_torque_rd_sr(adcs::rwa::min_torque, adcs::rwa::max_torque, 16*3), //referenced from I2C_Interface.doc
+    rwa_torque_rd_component_sr(adcs::rwa::min_torque,adcs::rwa::max_torque, 8),
```

Comment:
Same deal here with the torque read component if we want full resolution:
```suggestion
    rwa_torque_rd_component_sr(adcs::rwa::min_torque,adcs::rwa::max_torque, 12),
```